### ISSUE TITLE
Executing a trade run after a database update gives an error. In my TD\data folder the Tradedangerous.prices file is not showing

### ISSUE THREAD

**Issue by rmb4253** [eyeonus/Trade-Dangerous#139]
Running this command 

c:\TDH>trade run --fr="Sounti/Teng Station" --cap=114 --cr=1111111111 --ly=16 --empty=21 --gpt=400 --jum=2 --lsp=12.5 --ls-max=12000 --prune-hops=2 --prune-score=90 --show-jumps --avoid=Slaves,Landmines --fc=N --pad=L -vvv --hops=4 --progress
C:\Python\Scripts\trade: Error: There is no trading data for ANY station in the local database. Please enter or import price data.
Possible causes:
- No profitable runs or routes meet your criteria,
- Missing Systems or Stations along the route (see "local -vv"),
- Missing price data (see "market -vv" or "update -h"),

If you are not sure where to get data from, consider using a crowd-sourcing
project such as EDDBlink (https://github.com/eyeonus/EDDBlink).

For more help, see the TradeDangerous Wiki:
    https://github.com/eyeonus/Trade-Dangerous/wiki

**Comment by eyeonus** [comment]
I wasn't able toreproduce the problem:
```
[eyeonus@panther TDtest]$ trade run --fr="Sounti/Teng Station" --cap=114 --cr=1111111111 --ly=16 --empty=21 --gpt=400 --jum=2 --lsp=12.5 --ls-max=12000 --prune-hops=2 --prune-score=90 --show-jumps --avoid=Slaves,Landmines --fc=N --pad=L -vvv --hops=4 --progress
* Hop   1: .........1 origins 
NOTE: Pruned 661 origins     
* Hop   2: ........74 origins .. 322,848-1,223,904cr gain, 2,832-10,736cr/ton
NOTE: Pruned 2692 origins    
* Hop   3: .......300 origins .. 1,562,598-3,433,338cr gain, 6,853-15,058cr/ton
NOTE: Pruned 5421 origins    
* Hop   4: .......603 origins .. 3,230,532-5,166,594cr gain, 9,446-15,107cr/ton
Sounti/Teng Station -> Votama/Clifford Astrophysics Enterprise (score: 6324430.217154)
Start CR: 1,111,111,111
Hops    :          4
Jumps   :          7
Gain CR :  6,114,960
Gain/Hop:  1,528,740
Final CR: 1,117,226,071

  Load from Sounti/Teng Station (79ls, BMk:N, Pad:L, Plt:N, Flc:N, Ody:N, Shp:Y, Out:Y, Ref:Y):
      114 x Medicines/Basic Medicines      245cr vs    5,001cr, 7 days vs 6 days, total:     27,930cr
  Jump Sounti, 5.90ly -> Mahiko
  Unload at Mahiko/Herrera Metallurgic Hub (2.14Kls, BMk:Y, Pad:L, Plt:Y, Flc:N, Ody:Y, Shp:N, Out:N, Ref:Y) => Gain 542,184cr (4,756cr/ton) => 1,111,653,295cr
  Load from Mahiko/Herrera Metallurgic Hub (2.14Kls, BMk:Y, Pad:L, Plt:Y, Flc:N, Ody:Y, Shp:N, Out:N, Ref:Y):
      114 x Minerals/Pyrophyllite        5,701cr vs   28,036cr, 6 days vs 20 hrs, total:    649,914cr
  Jump Mahiko, 15.37ly -> LTT 3787, 14.36ly -> HIP 50694
  Direct: 26.44ly, Trip: 29.73ly
  Unload at HIP 50694/Zebrowski Enterprise (930ls, BMk:N, Pad:L, Plt:N, Flc:N, Ody:N, Shp:Y, Out:Y, Ref:Y) => Gain 2,546,190cr (22,335cr/ton) => 1,114,199,485cr
  Load from HIP 50694/Zebrowski Enterprise (930ls, BMk:N, Pad:L, Plt:N, Flc:N, Ody:N, Shp:Y, Out:Y, Ref:Y):
      114 x Medicines/Basic Medicines      447cr vs    5,001cr, 20 hrs vs 6 days, total:     50,958cr
  Jump HIP 50694, 14.36ly -> LTT 3787, 15.37ly -> Mahiko
  Direct: 26.44ly, Trip: 29.73ly
  Unload at Mahiko/Herrera Metallurgic Hub (2.14Kls, BMk:Y, Pad:L, Plt:Y, Flc:N, Ody:Y, Shp:N, Out:N, Ref:Y) => Gain 519,156cr (4,554cr/ton) => 1,114,718,641cr
  Load from Mahiko/Herrera Metallurgic Hub (2.14Kls, BMk:Y, Pad:L, Plt:Y, Flc:N, Ody:Y, Shp:N, Out:N, Ref:Y):
      114 x Minerals/Pyrophyllite        5,701cr vs   27,696cr, 6 days vs 49 days, total:    649,914cr
  Jump Mahiko, 12.83ly -> Riki, 11.98ly -> Votama
  Direct: 19.39ly, Trip: 24.80ly
  Unload at Votama/Clifford Astrophysics Enterprise (22ls, BMk:N, Pad:L, Plt:Y, Flc:N, Ody:Y, Shp:N, Out:N, Ref:Y) => Gain 2,507,430cr (21,995cr/ton) => 1,117,226,071cr
  ----------------------------------------------------------------------------
Finish at Votama/Clifford Astrophysics Enterprise (22ls, BMk:N, Pad:L, Plt:Y, Flc:N, Ody:Y, Shp:N, Out:N, Ref:Y) gaining 6,114,960cr (13,410cr/ton) => est 1,117,226,071cr total

```

Would you please do `trade import -O clean` and try the run again?

**Comment by Tromador** [comment]
> C:\Python\Scripts\trade: Error: There is no trading data for ANY station in the local database. Please enter or import price data. 

It's likely you've not rebuilt the database since some recent database breaking changes (basically switching all the old TD id numbers to match FDev ID numbers. 

As eyenus says above, a clean run will likely fix the problem, however it might be better to completely clean out the TD data folder, to ensure you've not got any old csv files which could cause confusion as you'll have to rebuild the database in any event. 

And eyeonus made a derp, he meant for you to run ```trade import -P eddblink -O clean``` but we won't hold it against him. :)

**Comment by rmb4253** [comment]
Good morning guys.  I have completely cleared out my TD\data folder and run the clean update.  That seemed to run as normal although just short of 2 hours to complete.  I copied the trade run that eyeonus did above and no errors and I got a trade run.  My data folder now contains the normal csv files, tradedangerous.db of 6,310,776 KB, tradedangerous.old of 144KB and a tradedangerous.sql of 8KB.  No sign of a tradedangerous.prices file so I guess it must be hidden away out of sight.  I can live with that so thanks for your help.

**Comment by rmb4253** [comment]
I've restarted my PC since the previous comment and started up TD again.  I looked a nearby system to see if any stations there needed a prices update and got this.

c:\TDAlone>trade local --ly=1 --pad=L -vvv "LTT 5131"
NOTE: Rebuilding cache file: this may take a few moments.
NOTE: Missing "C:\TDAlone\data\TradeDangerous.prices" file - no price data.
 System      Dist
  /   Station........................ StnLs Age/days Mkt BMk Shp Out Arm Ref Rep Pad Plt Flc Ody Itms
-----------------------------------------------------------------------------------------------------
 LTT 5131    0.00
  /   Spring Hub..................... 2.03K        - Yes  No Yes Yes Yes Yes Yes Lrg  No  No  No    0
  /   V0F-TQF........................ 2.21K        - Yes  No  No  No Yes Yes Yes Lrg  No Yes  No    0
  /   Osnovianenko's Research........ 3.33K        - Yes  No  No  No  No Yes Yes Lrg Yes  No Yes    0
  /   Brazier Chemical Enterprise.... 2.21K        - Yes  No  No  No  No Yes Yes Lrg Yes  No Yes    0
  /   Alvarez's Astrophysics......... 3.33K        - Yes Yes  No  No  No Yes Yes Lrg Yes  No Yes    0
  /   Caraiman Analytics Laboratory.. 2.21K        - Yes  No  No  No  No Yes Yes Lrg Yes  No Yes    0
  /   Dara Synthetics Installation... 3.36K        - Yes Yes  No  No  No Yes Yes Lrg Yes  No Yes    0
  /   Figueroa Engineering Foundry... 3.36K        - Yes  No  No  No  No Yes Yes Lrg Yes  No Yes    0
  /   Cortes Forge................... 3.33K        - Yes Yes  No  No  No Yes Yes Lrg Yes  No Yes    0
  /   Sargent's Industrial........... 3.34K        - Yes Yes  No  No  No Yes Yes Lrg Yes  No Yes    0
  /   Hidalgo Industrial Works....... 2.23K        - Yes  No  No  No  No Yes Yes Lrg Yes  No Yes    0

Uh-oh! No prices file.  Reading the comment by eyeonus about downloading the prices file, I ran this 
c:\TDAlone>trade import -P eddblink -O listings,prices
NOTE: Checking for update to 'System.csv'.
NOTE: Checking for update to 'Station.csv'.
NOTE: Checking for update to 'Item.csv'.
NOTE: Checking for update to 'listings.csv'.
NOTE: Checking for update to 'listings-live.csv'.
NOTE: Downloading file 'listings-live.csv'.
NOTE: Requesting https://elite.tromador.com/files/listings-live.csv
NOTE: Downloaded  13.4MB of gziped data  15.9MB/s
NOTE: Processing market data from listings-live.csv: Start time = 2024-04-27 13:08:21.955795. Live = True
NOTE: Optimizing database...
NOTE: Finished processing market data. End time = 2024-04-27 13:08:41.894473
NOTE: Regenerating .prices file.
NOTE: Import completed.

This generated a prices file but only about 2MB.  I guess as it only downloaded the listings-live.csv file that was all it had to work on.  

As expected, repeating my earlier trade run command ,
c:\TDAlone>trade run --fr="Sounti/Teng Station" --cap=114 --cr=1111111111 --ly=16 --empty=21 --gpt=400 --jum=2 --lsp=12.5 --ls-max=12000 --prune-hops=2 --prune-score=90 --show-jumps --avoid=Slaves,Landmines --fc=N --pad=L -vvv --hops=4 --progress
C:\Python\Scripts\trade: Error: No price data in local database for --from station: Sounti/Teng Station
Possible causes:
- No profitable runs or routes meet your criteria,
- Missing Systems or Stations along the route (see "local -vv"),
- Missing price data (see "market -vv" or "update -h")

So no prices for that system.  I'm now running another clean download and will follow that with the prices download and experiment some more.
